### PR TITLE
Development

### PR DESCRIPTION
**Pull request by DerCalli**
**Changelog**

- rename DataLayerExpander to StoreDataLayerExpander
- add system field for datalayer
- some code fixes

### CODE REVIEW

File: src/FondOfSpryker/Yves/GoogleTagManagerStoreConnector/Expander/StoreDataLayerExpander.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    /**
+     * @return string
+     */
+    protected function getEnviroment(): string
+    {
+        if (!defined(APPLICATION_ENV)) {
+            return '';
+        }
+
+        if (APPLICATION_ENV === GoogleTagManagerStoreConnectorConstants::ENV_PRODUCTION) {
```
Comment (julianhyatt) [inline src/FondOfSpryker/Yves/GoogleTagManagerStoreConnector/Expander/StoreDataLayerExpander.php:96]: makes it more sense to use the normal env as return we use in our stack to keep it same? You already use it in const too

File: src/FondOfSpryker/Yves/GoogleTagManagerStoreConnector/Expander/StoreDataLayerExpander.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        GoogleTagManagerStoreConnectorConfig $config
+    ) {
+        $this->storeClient = $storeClient;
+        $this->config = $config;
+    }
+
+    /**
+     * @param string $page
+     * @param array $twigVariableBag
+     * @param array $dataLayer
```
Comment (gengjozsef) [inline src/FondOfSpryker/Yves/GoogleTagManagerStoreConnector/Expander/StoreDataLayerExpander.php:36]: you could also show the type of objects store in array,. Example string[]

**Review by julianhyatt — APPROVED** [review]

File: src/FondOfSpryker/Yves/GoogleTagManagerStoreConnector/Expander/StoreDataLayerExpander.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+     *
+     * @return array
```
Comment (gengjozsef) [inline src/FondOfSpryker/Yves/GoogleTagManagerStoreConnector/Expander/StoreDataLayerExpander.php:38]: by the returned array you could also show the type of objects container from array if it is possible

File: src/FondOfSpryker/Yves/GoogleTagManagerStoreConnector/Expander/StoreDataLayerExpander.php [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+     * @return string
+     */
+    protected function getEnviroment(): string
+    {
+        if (!defined(APPLICATION_ENV)) {
+            return '';
+        }
+
+        if (APPLICATION_ENV === GoogleTagManagerStoreConnectorConstants::ENV_PRODUCTION) {
+            return 'Prod';
```
Comment (gengjozsef) [inline src/FondOfSpryker/Yves/GoogleTagManagerStoreConnector/Expander/StoreDataLayerExpander.php:97]: you could use constants

File: src/FondOfSpryker/Yves/GoogleTagManagerStoreConnector/Expander/StoreDataLayerExpander.php [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        }
+
+        if (APPLICATION_ENV === GoogleTagManagerStoreConnectorConstants::ENV_STAGING) {
+            return 'Stage';
```
Comment (gengjozsef) [inline src/FondOfSpryker/Yves/GoogleTagManagerStoreConnector/Expander/StoreDataLayerExpander.php:101]: the same constants

**Review by gengjozsef — APPROVED** [review]

**Pull request merged**